private/bufpkg/bufcheck/buflint/internal/buflintcheck/buflintcheck.go

```diff
@@ -963,3 +968,95 @@ func checkSyntaxSpecified(add addFunc, file protosource.File) error {
 	}
 	return nil
 }
+
+var (
+	// fieldConstraintsDesc provides a Descriptor for validate.FieldConstraints.
+	fieldConstraintsDesc = (*validate.FieldConstraints)(nil).ProtoReflect().Descriptor()
+
+	// fieldConstraintsOneOfDesc provides the OneOfDescriptor for the type union
+	// in FieldConstraints.
+	fieldConstraintsOneOfDesc = fieldConstraintsDesc.Oneofs().ByName("type")
```

Comment:
Is this not better named `fieldConstraintsTypeOneofDesc`?

Note: fixed this in a push.